videoretrieval/helper/precomputed_features.py

```diff
@@ -0,0 +1,107 @@
+"""Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Functions to download, extract, and process precomputed features.
+"""
+
+from helper import file_downloader, tar_helper
+from cache import features_cache
+import pickle
+
+def download_features_tar(features_tar_url, features_tar_path):
+    """Downloads a tar file of precomputed features.
+
+    Arguments:
+        features_tar_url: a string that is the url of the features tar.
+        features_tar_path: the path the tar should be saved to.
+    """
+
+    features_tar_path.parent.mkdir(parents=True, exist_ok=True)
+
+    file_downloader.download_by_url(
+        features_tar_url,
+        features_tar_path
+    )
+
+def extract_features_from_file(file):
+    """Extracts features from a file using `pickle.load`."""
+    return pickle.load(file)
+
+def verify_features_dimensioality(features, expert):
+    """Checks that the dimensionality of the embeddings matches expectations."""
```

Comment:
Does anything happen when this check fails? Looks like it does not return anything but just prints.